src/code/z_actor.c

```diff
@@ -456,232 +469,236 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
     targetCtx->fairyOuter.a = sTatlColorList[type].outer.a;
 }
 
-void Actor_TargetContextInit(TargetContext* targetCtx, Actor* actor, PlayState* play) {
+void Target_Init(TargetContext* targetCtx, Actor* actor, PlayState* play) {
     targetCtx->bgmEnemy = NULL;
-    targetCtx->unk8C = NULL;
+    targetCtx->unk_8C = NULL;
     targetCtx->targetedActor = NULL;
     targetCtx->arrowPointedActor = NULL;
-    targetCtx->unk4B = 0;
-    targetCtx->unk4C = 0;
+    targetCtx->rotation = 0;
+    targetCtx->currentLockOnIndex = 0;
     targetCtx->unk40 = 0.0f;
     Target_SetColors(targetCtx, actor, actor->category, play);
-    func_800B4F78(targetCtx, actor->category, play);
+    Target_InitLockOn(targetCtx, actor->category, play);
 }
 
-void Actor_DrawZTarget(TargetContext* targetCtx, PlayState* play) {
+void Target_Draw(TargetContext* targetCtx, PlayState* play) {
     Player* player = GET_PLAYER(play);
+    Actor* actor;
 
-    if (!(player->stateFlags1 & (PLAYER_STATE1_2 | PLAYER_STATE1_40 | PLAYER_STATE1_80 | PLAYER_STATE1_200 |
-                                 PLAYER_STATE1_400 | PLAYER_STATE1_10000000 | PLAYER_STATE1_20000000))) {
-        Actor* actor = targetCtx->targetedActor;
+    if (player->stateFlags1 & (PLAYER_STATE1_2 | PLAYER_STATE1_40 | PLAYER_STATE1_80 | PLAYER_STATE1_200 |
+                               PLAYER_STATE1_400 | PLAYER_STATE1_10000000 | PLAYER_STATE1_20000000)) {
+        return;
+    }
 
-        OPEN_DISPS(play->state.gfxCtx);
+    actor = targetCtx->targetedActor;
 
-        if (targetCtx->unk48 != 0) {
-            TargetContextEntry* entry;
-            s16 alpha = 255;
-            f32 var1 = 1.0f;
-            Vec3f projectedPos;
-            s32 spB8;
-            f32 invW;
-            s32 spB0;
-            s32 spAC;
-            f32 var2;
-            s32 i;
-
-            if (targetCtx->unk4B != 0) {
-                spB8 = 1;
-            } else {
-                spB8 = 3;
-            }
+    OPEN_DISPS(play->state.gfxCtx);
 
-            if (actor != NULL) {
-                Math_Vec3f_Copy(&targetCtx->targetCenterPos, &actor->focus.pos);
-                var1 = (500.0f - targetCtx->unk44) / 420.0f;
-            } else {
-                targetCtx->unk48 -= 120;
-                if (targetCtx->unk48 < 0) {
-                    targetCtx->unk48 = 0;
-                }
-                alpha = targetCtx->unk48;
+    if (targetCtx->lockOnAlpha != 0) {
+        TargetLockOnEntry* entry;
+        s16 alpha = 255;
+        f32 var1 = 1.0f;
+        Vec3f projectedPos;
+        s32 totalEntries;
+        f32 invW;
+        s32 i;
+        s32 index;
+        f32 var2;
+
+        if (targetCtx->rotation != 0) {
+            totalEntries = 1;
+        } else {
+            // Use multiple entries for movement effect when the are getting closer to the actor from the margin of the
+            // screen
+            totalEntries = ARRAY_COUNT(targetCtx->lockOnEntries);
```

Comment:
```suggestion
            // Use multiple entries for movement effect when they are getting closer to the actor from the margin of the
            // screen
            totalEntries = ARRAY_COUNT(targetCtx->lockOnEntries);
```
they -> the triangles? the player?

Reply:
The triangles, yes :sweat_smile: